### PR TITLE
[12.0][MIG] account_permanent_lock_move

### PR DESCRIPTION
**Pull request by elouan-lb**
Standard migration procedure, tested locally.

This migration is based on the work of @sbejaoui in #830

### CODE REVIEW

**Review by houssine78 — APPROVED** [review]

**Comment by houssine78** [comment]
For info, we use it in production for our client. Working pretty well so far

File: account_permanent_lock_move/tests/test_permanent_lock.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                'lock_date': fields.Date.today()
+            })
+            self.wizard.confirm_date()
+        except UserError as ue:
+            if 'entries are still unposted' in ue.name:
+                raised_lock_error = True
+
+        self.assertTrue(raised_lock_error,
+                        "Permanent lock done even with unposted entry.")
+
```
Comment (fcayre) [inline account_permanent_lock_move/tests/test_permanent_lock.py:56]: You'd better use the dedicated context manager, with (untested) code like:

```
with self.assertRaises(UserError) as exc:
   self.wizard = self.wizard_obj.create({
       'company_id': self.company_id,
       'lock_date': fields.Date.today()
    })
    self.wizard.confirm_date()
self.assertIn('entries are still unposted', str(exc.err))
```

File: account_permanent_lock_move/wizard/permanent_lock_date_wizard.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                 string='Company')
+
+    @api.multi
+    def confirm_date(self):
+        self.ensure_one()
+        company = self.company_id
+        if (company.permanent_lock_date and
+                self.lock_date < company.permanent_lock_date):
+            raise UserError(
+                _("You cannot set the permanent lock date in the past.")
```
Comment (fcayre) [inline account_permanent_lock_move/wizard/permanent_lock_date_wizard.py]: I do not understand this test. You may safely ignore the remark as I am not a functional guy, but self.lock_date is not tested against present date here, so the message error message mentioning "in the past" looks wrong (if at least company.permanent_lock_date can be in the future, which looks possible to me here). Maybe this deserves a bit more precise documentation or a method docstring.

**Review by fcayre — APPROVED** [review]
Code review only, with non-blocking remarks.

File: account_permanent_lock_move/tests/test_permanent_lock.py (hunk 1 above)
Comment (houssine78) [reply account_permanent_lock_move/tests/test_permanent_lock.py:56]: done

File: account_permanent_lock_move/wizard/permanent_lock_date_wizard.py (hunk 2 above)
Comment (houssine78) [reply account_permanent_lock_move/wizard/permanent_lock_date_wizard.py]: done

**Review by robinkeunen — APPROVED** [review]
LGTM 👌

File: account_permanent_lock_move/tests/test_permanent_lock.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                'lock_date': yesterday
+            })
+            self.wizard.confirm_date()
+        except UserError as ue:
+            if 'permanent lock date in the past' in ue.name:
+                raised_lock_error = True
+
+        self.assertTrue(raised_lock_error,
+                        "Permanent lock set the day before.")
+
```
Comment (robinkeunen) [inline account_permanent_lock_move/tests/test_permanent_lock.py]: Same remark here, use assert raise. Travis raised an error here

**Comment by houssine78** [comment]
Hi @pedrobaeza this PR is approved.

But I can't get the CI passing. I get a weird lint error. And Runbot doesn't give me any hit why it's failing. If you're are king enough to point me to the problem, I'll be glad to get them fixed.

We use this module in production for several months now.

**Comment by pedrobaeza** [comment]
Travis errors are:
```
account_permanent_lock_move/tests/test_permanent_lock.py:52:16: E121 continuation line under-indented for hanging indent
account_permanent_lock_move/tests/test_permanent_lock.py:78:14: E121 continuation line under-indented for hanging indent
account_permanent_lock_move/tests/test_permanent_lock.py:100:46: F841 local variable 'exc' is assigned to but never used
```
https://travis-ci.com/github/OCA/account-financial-tools/jobs/449891428#L324

Runbot is simply being killed by other reasons not related to this PR.

**Comment by houssine78** [comment]
@pedrobaeza thanks for the hints ! It's green and ready to be merge !

thanks you !

**Comment by robinkeunen** [comment]
@pedrobaeza I think this time, we're good to go :-)

**Pull request merged**